lib/gosu/compat.rb

```diff
@@ -101,6 +101,70 @@ def self.from_text(*args)
       from_text_without_window(*args)
     end
   end
+
+  alias _draw_rot draw_rot
+  def draw_rot(*args)
+    Gosu.deprecation_message("Image#draw_rot is deprecated in Gosu x.y, use draw with an options hash instead, see https://www.libgosu.org/rdoc/Gosu/Image.html.")
+    _draw_rot(*args)
+  end
+
+  alias _draw draw
+  DRAW_ALIGNMENTS = {
+    :upper_left => [-1, -1],
+    :up => [0, -1],
+    :upper_right => [1, -1],
+    :left => [-1, 0],
+    :middle => [0, 0],
+    :right => [1, 0.0],
+    :bottom_left => [-1, 1],
+    :bottom => [0, 1],
+    :bottom_right => [1, 1],
+  }
+  def draw(x, y, z, *args)
+    if args.last.is_a?(Hash)
+      opts = args.pop
+      scale_x  = args[0] || opts.delete(:scale_x) || 1
+      scale_y  = args[1] || opts.delete(:scale_y) || 1
+      color    = args[2] || opts.delete(:color) || 0xff_ffffff
+      mode     = args[3] || opts.delete(:mode) || :default
+      center_x = opts.delete(:center_x)
+      center_y = opts.delete(:center_y)
+
+      # Shorthand for scale_x: scale_y:
+      scale = opts.delete(:scale)
+      scale_x, scale_y = scale, scale if scale
+
+      # Shortcut with symbols
+      center = opts.delete(:center)
+      if center
+        center_x, center_y = if center.is_a?(Symbol)
+          if DRAW_ALIGNMENTS.has_key?(center)
+            DRAW_ALIGNMENTS[center]
+          else
+            raise ArgumentError, "Unknown drawing alignmengt: #{center} should be a float or one of: #{DRAW_ALIGNMENTS.keys.join(', ')}"
+          end
+        else
+          [center, center]
+        end
+      end
+
+      # Make draw_rot obsolete as the params are basically the same besides the rotation-specific values
+      angle = opts.delete(:angle)
+
+      raise ArgumentError, "Unknown options: #{opts.keys} should be either of [angle, center, center_x, center_y, color, mode, scale, scale_x, scale_y]" unless opts.empty?
+      if angle or center_x or center_y
```

Comment:
I don't think it's intuitive that passing `center_x: 0.5` also sets `center_y: 0.5`, but leaving out `center_x` uses `center_y: 0`

Reply:
I still think changing the default value of `center_x`/`center_y` based on the presence of `angle` is more intuitive, if only slightly :)

Reply:
You're right. Was a bit confused as I need to call draw_rot if any of the 3 is set because draw doesn't support center so I had to set angle to a default as well but I guess now I got it ;-) but not sure how you came to 

> out center_x uses center_y: 0

all defaulted to 0.5

Reply:
It only defaults to 0.5 when `angle` is given. So:

```ruby
image.draw x, y, z, center_x: 0.5 # => center_y = 0.5 (surprising)
image.draw x, y, z => center_y = 0.0 (OK)
```